server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -557,7 +558,12 @@ public void messageReceived(FieldCapabilitiesNodeRequest request, TransportChann
                     .stream()
                     .collect(Collectors.groupingBy(ShardId::getIndexName));
                 final FieldCapabilitiesFetcher fetcher = new FieldCapabilitiesFetcher(indicesService, request.includeEmptyFields());
-                final Predicate<String> fieldNameFilter = Regex.simpleMatcher(request.fields());
+                Predicate<String> fieldNameFilter;
+                try {
+                    fieldNameFilter = Regex.simpleMatcher(request.fields());
+                } catch (TooComplexToDeterminizeException e) {
+                    throw new IllegalArgumentException("The field names are too complex to process. " + e.getMessage());
```

Comment:
Could you add a yaml test to make sure this comes out as a nice 400? I'm pretty sure it does, but those tests are mostly to make sure we don't break it in nine months.

Reply:
Added a test in `RestEsqlTestCase`.